tbb 2021.10.0 builds without using swapcontext, but we can just enable it again
and link libucontext for resumable support
--
diff --git a/cmake/compilers/GNU.cmake b/cmake/compilers/GNU.cmake
index b60172c..98eb9f2 100644
diff --git a/include/oneapi/tbb/detail/_config.h b/include/oneapi/tbb/detail/_config.h
index ad9f0f3..31e7de7 100644
--- a/include/oneapi/tbb/detail/_config.h
+++ b/include/oneapi/tbb/detail/_config.h
@@ -274,7 +274,7 @@
     #define __TBB_CPP20_COMPARISONS_PRESENT 0
 #endif
 
-#define __TBB_RESUMABLE_TASKS                           (!__TBB_WIN8UI_SUPPORT && !__ANDROID__ && !__QNXNTO__ && (!__linux__ || __GLIBC__))
+#define __TBB_RESUMABLE_TASKS                           (!__TBB_WIN8UI_SUPPORT && !__ANDROID__ && !__QNXNTO__ && (__linux__))
 
 /* This macro marks incomplete code or comments describing ideas which are considered for the future.
  * See also for plain comment with TODO and FIXME marks for small improvement opportunities.
--- a/cmake/compilers/Clang.cmake
+++ b/cmake/compilers/Clang.cmake
@@ -54,7 +54,7 @@
 	                                                  -fstack-protector-strong -fPIC)
 set(TBB_LIB_LINK_FLAGS ${TBB_LIB_LINK_FLAGS} -Wl,-z,relro,-z,now)
 
-set(TBB_COMMON_LINK_LIBS ${CMAKE_DL_LIBS})
+set(TBB_COMMON_LINK_LIBS ${CMAKE_DL_LIBS} ucontext)
 
 set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} $<$<NOT:$<CONFIG:Debug>>:-D_FORTIFY_SOURCE=2>)
 
